server/activityItem/queries.ts

```diff
@@ -0,0 +1,340 @@
+import * as types from 'types';
+
+import {
+	Collection,
+	Member,
+	Community,
+	PubEdge,
+	Discussion,
+	ExternalPublication,
+	ReviewNew,
+	Pub,
+	CollectionPub,
+	ThreadComment,
+	Thread,
+} from 'server/models';
+
+import { getDiffsForPayload, getChangeFlagsForPayload, createActivityItem } from './utils';
+
+export const createCommunityActivityItem = async (
+	kind: 'community-created' | 'community-updated',
+	actorId: string,
+	oldCommunity: types.Community,
+	communityId: string,
+) => {
+	const community: types.Community = await Community.findOne({ where: { id: communityId } });
+	const diffs = getDiffsForPayload(community, oldCommunity, ['title']);
+	return createActivityItem({
+		actorId,
+		kind,
+		communityId,
+		payload: {
+			...diffs,
+			community: {
+				title: community.title,
+			},
+		},
+	});
+};
+
+export const createMemberUpdatedActivityItem = async (
+	kind: 'member-updated',
+	actorId: string,
+	communityId: string,
+	memberId: string,
+	oldMember: types.Member,
+) => {
+	const member: types.Member = await Member.findOne({ where: { id: actorId } });
+	const diffs = getDiffsForPayload(member, oldMember, ['permissions']);
+	return createActivityItem({
+		kind,
+		actorId,
+		communityId,
+		payload: {
+			userId: memberId,
```

Comment:
We want the ID of the `User` associated with the `Member` object — so `member.userId`